api.ts

```diff
@@ -68,21 +80,45 @@ interface StatusCodeErrorResponse {
   headers?: {[key: string]: string | string[] | undefined};
 }
 
+// StatusCodeError is a simple version of StatusCodeError in request-promise to keep backwards compatibility.
+// This tries to replicate its exact structure, massaging as necessary to handle the various transforms
+// in our stack.
+//
+// https://github.com/request/promise-core/blob/master/lib/errors.js#L22
 /**
- * StatusCodeError is a simple version of StatusCodeError in request-promise to keep backwards compatibility.
- * This tries to replicate its exact structure, massaging as necessary to handle the various transforms
- * in our stack.
+ * An error that will be thrown by {@link Fetcher.fetch} when the fetcher response has an
+ * HTTP status code of 400 or greater.
  *
- * https://github.com/request/promise-core/blob/master/lib/errors.js#L22
+ * This class largely models the `StatusCodeError` from the (now deprecated) `request-promise` library,
+ * which has a quirky structure.
  */
 export class StatusCodeError extends Error {
+  /**
+   * The name of the error, for identiciation purposes.
+   */
   override name: string = 'StatusCodeError';
+  /**
+   * The HTTP status code, e.g. `404`.
+   */
   statusCode: number;
+  /**
+   * The raw body of the HTTP response.
```

Comment:
In my testing for the [fetcher guide](https://coda.github.io/packs-sdk/guides/advanced/fetcher/#errors) I found that this contains the parsed body, just like `response.body`. If this is WAI then I think "raw" gives the wrong impression.

Reply:
Ah good to know. Yeah this whole object is confusing (the confusion comes from the request-promise behavior we're trying to keep compatibility with). That sounds right. In which case, I think `error` is just an alias for `body`. I believe `response.body` is supposed to be unparsed, and indeed it looks like we try to ensure that it is: https://github.com/coda/packs-sdk/blob/main/api.ts#L98

Reply:
I found it extra confusing that `response.body` is parsed, but `error.response.body` isn't parsed. Is the compatibility with request-promise still needed, or is that legacy?

Reply:
We have a bunch of old packs relying on this. Ideally we would fix them but it's pretty hard to track down all of the obscure behavior that they're relying on, in particular because of the weak typing.